metricbeat/module/elasticsearch/index/data.go

```diff
@@ -285,6 +305,34 @@ func getClusterStateMetricForIndex(clusterState mapstr.M, index, metricKey strin
 	return mapstr.M(metric), nil
 }
 
+func getIndexTierPreferenceFromMetadata(indexMetadata mapstr.M) (string, error) {
+	fieldKey := "settings.index.routing.allocation.include._tier_preference"
```

Comment:
In addition to `include. _tier_preference` I would also consider `require. _tier_preference` since that's also a valid setting